src/Streamly/Internal/Data/Parser/ParserD/Types.hs

```diff
@@ -145,66 +145,69 @@ import Streamly.Internal.Data.Strict (Tuple3'(..))
+-- of the parser to extract a result. If the parser has not yet yielded a
+-- result, the operation fails with a 'ParseError' exception. If the parser
+-- yielded a 'Partial' result in the past the last partial result is returned.
+-- Therefore, if a parser yields a partial result once it cannot fail later on.
+--
+-- The parser can never backtrack beyond the position where the last partial
+-- result left it at. The parser must ensure that the backtrack position is
+-- always after that.
 --
 -- /Internal/
 --
 {-# ANN type Step Fuse #-}
 data Step s b =
-      Yield Int s
-      -- ^ Trim the backtracking buffer keeping only most recent @n@ items.
-      -- @Yield n state@ indicates that the parser has yielded a new result
-      -- which is a point of no return. The result can be extracted using
-      -- @extract@. The driver drops the buffer except @n@ most recent
-      -- elements.  The rule is that if a parser has yielded at least once it
-      -- cannot return a failure result.
-
-      | YieldB Int s
-      -- ^ Go back by @n@ items in the input buffer and drop the rest.
-      -- @YieldB n state@ indicates that the parser has yielded a new result
-      -- which is a point of no return. The result can be extracted using
-      -- @extract@. The driver moves back the current position of the cursor by
-      -- @n@ elements and drops any buffer before that.
-
-     -- only rewind
-    | Skip Int s
-    -- ^ Go back by @n@ items in the input buffer.
-    -- @Skip n state@ indicates that the parser has consumed the current input
-    -- but no new result has been generated. A new @state@ is generated.
-    -- However, if we use @extract@ on @state@ it will generate a result from
-    -- the previous @Yield@.  When @n@ is non-zero it is a backward offset from
-    -- the current position in the stream from which the driver will replay the
-    -- input to the parser. The offset cannot be beyond the latest point of no
-    -- return created by @Yield@.
-
-    | Stop Int b
-    -- ^ Trim the backtracking buffer keeping only most recent @n@ items.
-    -- @Stop n result@ asks the driver to stop driving the parser because it
-    -- has reached a fixed point and further input will not change the result.
-    -- @n@ is the count of unused input elements which includes the element on
-    -- which 'Stop' occurred.
+        Partial Int s
+    -- ^ Partial result with an optional backtrack.
+    --
+    -- @PartialWith count state@ means a partial result is available which
+    -- can be extracted successfully, @state@ is the opaque state of the
+    -- parser to be supplied to the next invocation of the step operation.
+    -- The current input position is reset to @count@ elements back and any
+    -- input before that is dropped from the backtrack buffer.
+
+    | Continue Int s
+    -- ^ Need more input with an optional backtrack.
+    --
+    -- @Continue n state@ means the parser has consumed the current input but
```

Comment:
Change `Continue n state` to `Continue count state` or change `count` to `n` in the following description.

Reply:
done.